incubator/hnc/api/v1alpha1/hierarchy_types.go

```diff
@@ -208,4 +224,12 @@ func SortAffectedObjects(objs []AffectedObject) {
 
 func init() {
 	SchemeBuilder.Register(&HierarchyConfiguration{}, &HierarchyConfigurationList{})
+	ClearConditionCriteria = map[Code]ClearConditionCriterion{
+		CritParentMissing: CCCNone,
+		CritParentInvalid: CCCNone,
+		CritAncestor:      CCCNone,
+		HNSMissing:        CCCNone,
+		CannotPropagate:   CCCNotDescendant,
+		CannotUpdate:      CCCNotAncestor,
```

Comment:
It's a little hard to understand `CCCNotDescendant` and `CCCNotDescendant`. Also it's not clear *who* is not descendant or ancestor.

Either change the const name to e.g. `CCCAONotDescendant` or add comments in the const declarations in L58-59.

Reply:
Added docs, thanks for keeping me honest :D

Reply:
Nice doc!